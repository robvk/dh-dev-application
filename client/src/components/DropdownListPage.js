import React from "react";
import map from "lodash/map";

import NoResult from "./NoResult";
import Loading from "./Loading";

import "./DropdownListPage.css";

type Props = {
  list: Array<any>,
  renderItem: Function,
  startNumber: number,
  totalCount: number,
  loading: boolean
};

const DropdownListPage = (props: Props) => {
  const { loading, list, renderItem, totalCount, startNumber } = props;

  if (loading)
    return (
      <div className="dropdown-list">
        <Loading />
      </div>
    );

  if (!list || list.length === 0)
    return (
      <div className="dropdown-list">
        <NoResult />
      </div>
    );

  return (
    <div className="dropdown-list">
      {map(list, (item, index) => renderItem(item, index))}
      <div className="dropdown-list-bottom-text">{`${startNumber}-${startNumber +
        list.length -
        1} of ${totalCount} results`}</div>
    </div>
  );
};

export default DropdownListPage;
