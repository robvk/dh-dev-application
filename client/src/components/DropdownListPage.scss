@import "../appColors";

.dropdown-list {
  z-index: 1;
  width: 100%;
  border: 1px solid $primary-color;
  border-top: 0;
  box-sizing: border-box;
}

.dropdown-list-bottom-text {
  color: $disabled-color;
  text-align: center;
  width: 100%;
}
