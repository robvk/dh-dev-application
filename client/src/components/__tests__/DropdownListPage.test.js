import React from "react";
import ReactDOM from "react-dom";

import DropdownListPage from "../DropdownListPage";

const manyClients = Array(10).fill({
  id: "59a2afbbfc13ae278700044e",
  first_name: "Bartram",
  last_name: "Nassi",
  email: "bnassi2@goodreads.com",
  gender: "Male",
  photo: "https://robohash.org/aeablanditiis.bmp?size=50x50&set=set1",
  origin: "Egypt"
});

const testProps = {
  list: manyClients,
  renderItem: () => {},
  startNumber: 1,
  totalCount: 20,
  loading: false
};

it("renders correctly", () => {
  const wrapper = shallow(<DropdownListPage />);
  wrapper.setProps(testProps);
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly when loading", () => {
  const wrapper = shallow(<DropdownListPage />);
  wrapper.setProps({ ...testProps, loading: true });
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly when empty", () => {
  const wrapper = shallow(<DropdownListPage />);
  wrapper.setProps({ ...testProps, list: [], totalCount: 0 });
  expect(wrapper).toMatchSnapshot();
});
