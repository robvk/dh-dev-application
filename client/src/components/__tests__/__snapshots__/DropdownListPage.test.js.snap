// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dropdown-list"
>
  <div
    className="dropdown-list-bottom-text"
  >
    1-10 of 20 results
  </div>
</div>
`;

exports[`renders correctly when empty 1`] = `
<div
  className="dropdown-list"
>
  <NoResult />
</div>
`;

exports[`renders correctly when loading 1`] = `
<div
  className="dropdown-list"
>
  <Loading />
</div>
`;
